File: Data Structures & Algorithms/subsets-ii/submission-1.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        ans = []
        nums.sort()
        def back(sub:List[int], nums:List[int]):
            if len(nums)==0:
                ans.append(sub.copy())
                return 

            sub.append(nums[0])
            back(sub , nums[1:])
            sub.pop()
            i = 0
            while i<len(nums)-1 and nums[i]==nums[i+1]: i+=1
            back(sub,nums[i+1:])
        back([],nums)

        return ans

File: Data Structures & Algorithms/non-overlapping-intervals/submission-1.js
class Solution {
  /**
   * @param {number[][]} intervals
   * @return {number}
   */
  eraseOverlapIntervals(intervals) {
    if (intervals.length === 0) return 0;
    intervals.sort((a, b) => a[1] - b[1]);
    let ans = 0;
    let fe = intervals[0][1];
    for (let i = 1; i < intervals.length; i++) {
      let [s, e] = intervals[i];
      if (fe > s) {
        ans++;
      } else {
        fe = e;
      }
    }
    return ans;
  }
}
